pkg/service/helper.go

```diff
@@ -32,3 +32,22 @@ func diskAttachmentByVmAndDisk(connection *ovirtsdk.Connection, vmId string, dis
 		diskId: diskId,
 	}
 }
+
+func diskAttachmentByDisk(connection *ovirtsdk.Connection, diskId string) (*ovirtsdk.DiskAttachment, *ovirtsdk.DiskAttachmentService, error) {
```

Comment:
I think you shouldn't return both the attachment and service...
you should return the attachment because that is what the function is supposed to do.
Later you can use the attachment to update and get the service